ioctl/test_ioctl.c

```diff
@@ -0,0 +1,348 @@
+					out_data = (const void *)(out2);
+					break;
+			}
+			ioctl_setResponse(&msg, request, err, out_data);
+		}
+		msgRespond(port, &msg, rid);
+	}
+	return retval;
+}
+
+
+TEST_GROUP(ioctl);
+
+
+TEST_SETUP(ioctl)
+{
+}
+
+TEST_TEAR_DOWN(ioctl)
+{
+}
+
+
+TEST(ioctl, regular_file)
+{
+	/* try regular file, not special device file */
+	int fd = open(PATH, O_RDWR | O_CREAT | O_TRUNC);
+	TEST_ASSERT_NOT_EQUAL_INT(-1, fd);
+	errno = 0;
+	int ret = ioctl(fd, TIOCSIG, NULL);
+	TEST_ASSERT_EQUAL_INT(-1, ret);
+	// TEST_ASSERT_EQUAL_INT(EBADF, errno);
+}
+
+
+TEST(ioctl, wrong_fd)
+{
+	errno = 0;
+	int ret = ioctl(123, 0, NULL);
+	TEST_ASSERT_EQUAL_INT(-1, ret);
+	TEST_ASSERT_EQUAL_INT(EBADF, errno);
+}
+
+
+TEST(ioctl, no_data)
+{
+	int32_t flag;
+	int ret;
+
+	/* Send data to driver by value */
+	ret = ioctl(dd, TIOCSIG, NULL);
+	TEST_ASSERT_EQUAL_INT(0, ret);
+	TEST_ASSERT_EQUAL_INT(0, lseek(fd, 0, SEEK_SET));
+	TEST_ASSERT_EQUAL_INT(4, read(fd, &flag, 4));
+	TEST_ASSERT_EQUAL_INT32(1, flag);
+}
+
+
+TEST(ioctl, in_val)
+{
+	int32_t rdata, data = 14;
+	int ret;
+
+	/* Send data to driver by value */
+	ret = ioctl(dd, TIOCVAL, data);
+	TEST_ASSERT_EQUAL_INT(0, ret);
+	TEST_ASSERT_EQUAL_INT(0, lseek(fd, 0, SEEK_SET));
+	TEST_ASSERT_EQUAL_INT(4, read(fd, &rdata, 4));
+	TEST_ASSERT_EQUAL_INT32(14, rdata);
+}
+
+
+TEST(ioctl, data_in)
+{
+	int32_t rdata, data_in = 20;
+	int ret;
+
+	/* Send data to driver by pointer */
+	ret = ioctl(dd, TIOCPTR, &data_in);
+	TEST_ASSERT_EQUAL_INT(0, ret);
+	TEST_ASSERT_EQUAL_INT(0, lseek(fd, 0, SEEK_SET));
+	TEST_ASSERT_EQUAL_INT(4, read(fd, &rdata, 4));
+	TEST_ASSERT_EQUAL_INT32(20, rdata);
+}
+
+
+TEST(ioctl, data_in_big)
+{
+	/* Big data so that it is not copied inside message */
+	int8_t data_in[BIG_SIZE];
+	memset(data_in, 3, BIG_SIZE);
+	int8_t rdata[BIG_SIZE];
+	int ret;
+
+	/* Send data to driver by pointer , data big enough to not be copied by ioctl_pack directly into message */
+	ret = ioctl(dd, TIOCPTR2, data_in);
+	TEST_ASSERT_EQUAL_INT(0, ret);
+	TEST_ASSERT_EQUAL_INT(0, lseek(fd, 0, SEEK_SET));
+	TEST_ASSERT_EQUAL_INT(BIG_SIZE, read(fd, rdata, BIG_SIZE));
+	TEST_ASSERT_EQUAL_MEMORY(data_in, rdata, BIG_SIZE);
+}
+
+
+TEST(ioctl, data_out)
+{
+	int32_t data_out;
+	int ret;
+
+	/* Get data from driver */
+	ret = ioctl(dd, TIOCOUT, &data_out);
+	TEST_ASSERT_EQUAL_INT(0, ret);
+	TEST_ASSERT_EQUAL_INT32(15, data_out);
+}
+
+
+TEST(ioctl, data_out_big)
+{
+	/* Big data so that it is not copied inside message */
+	int8_t data_expected[BIG_SIZE];
+	memset(data_expected, 5, BIG_SIZE);
+	int8_t data_out[BIG_SIZE];
+	int ret;
+
+	/* Get data from driver */
+	ret = ioctl(dd, TIOCOUT2, data_out);
+	TEST_ASSERT_EQUAL_INT(0, ret);
+	TEST_ASSERT_EQUAL_MEMORY(data_expected, data_out, BIG_SIZE);
+}
+
+
+TEST_GROUP_RUNNER(ioctl)
+{
+	char devName[5] = "test";
+	oid_t dev;
+	int res, n = 0;
+
+	fd = open(PATH, O_RDWR | O_CREAT | O_TRUNC);
+	if (fd < 0) {
+		fprintf(stderr, "Couldn't open file\n");
+		return 1;
```

Comment:
I know that it's work in progress version, but this part shouldn't certainly be in `TEST_GROUP_RUNNER`
![Screenshot from 2023-05-31 12-54-19](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/77304431/07472dc5-f3ce-4fbd-911d-7822825dbd09)